src/graphics/shader.rs

```diff
@@ -113,6 +115,18 @@ impl From<BlendMode> for Blend {
                     destination: Factor::OneMinus(BlendValue::SourceAlpha),
                 },
             },
+            BlendMode::Premultiplied => Blend {
+                color: BlendChannel {
+                    equation: Equation::Sub,
+                    source: Factor::One,
+                    destination: Factor::ZeroPlus(BlendValue::SourceAlpha),
+                },
+                alpha: BlendChannel {
+                    equation: Equation::Sub,
+                    source: Factor::One,
+                    destination: Factor::One,
```

Comment:
This matters much less, but technically this should be:

```suggestion
                    equation: Equation::Add,
                    source: Factor::OneMinus(BlendValue::DestinationAlpha),
                    destination: Factor::One,
```

Reply:
Why exactly does this have to be ```(ONE_MINUS_DEST_ALPHA, ONE)``` instead of just ```(ONE, ONE)```, like how usual Alpha is implemented:
```rust
pub const ALPHA: Blend = Blend {
        color: BlendChannel {
            equation: Equation::Add,
            source: Factor::ZeroPlus(BlendValue::SourceAlpha),
            destination: Factor::OneMinus(BlendValue::SourceAlpha),
        },
        alpha: BlendChannel {
            equation: Equation::Add,
            source: Factor::One,
            destination: Factor::One,
        },
    };
```
I see why one might use ```(ONE_MINUS_DEST_ALPHA, ONE)``` instead of ```(ONE, ONE)```, since this is how transparency works in the real world. When you lay something that's half transparent (meaning half the light gets through) over something else that's half transparent then you'll get something where 75% of light passes, instead of something completely intransparent.
But I'm not yet fully convinced... wouldn't we need to change the definition of [ALPHA](https://docs.rs/draw_state/0.8.0/draw_state/preset/blend/constant.ALPHA.html) if this was true?

Thanks for helping btw.